site-manage/site-group-manager/src/main/webapp/WEB-INF/templates/fragments/javascript.html

```diff
@@ -0,0 +1,221 @@
+    });
+
+    if ($('#groupJoinableSet').val() == '') {
+      $('#joinableOptionsDiv').hide();
+    }
+
+    $('#groupJoinableSet').on('change', function() {
+      this.value != '' ? $('#joinableOptionsDiv').show() : $('#joinableOptionsDiv').hide();
+    });
+
+    //Initialize the multiselect component
+    var searchPlaceHolderText = /*[[#{groups.multiselect.search}]]*/;
+    $('#groupMembersFrom').multiselect({
+      sort: false,
+      startUp: true,
+      search: {
+        left: '<input type="text" name="q" class="form-control" placeholder="'+searchPlaceHolderText+'" />',
+        right: '<input type="text" name="q" class="form-control" placeholder="'+searchPlaceHolderText+'" />',
+      }
+    });
+
+  });
+
+  /*]]>*/
+</script>
+
+<script th:fragment="autoGroupsStep1Js" th:inline="javascript">
+  /*<![CDATA[*/
+  $('#selectAllRoles').click(function(e) {
+    var selectAllChecked = this.checked;
+    $(':checkbox').each(function() {
+      this.checked = selectAllChecked;
+    });
+  });
+  /*]]>*/
+</script>
+
+<script th:fragment="autoGroupsStep2Js" th:inline="javascript">
+  /*<![CDATA[*/
+  $("input[name='sectionsOption']").change(function(){
+    this.value === '0' ? $('#selectSectionDiv').fadeOut() : $('#selectSectionDiv').fadeIn();
+  });
+
+  $('#selectAllSections').click(function(e) {
+    var selectAllChecked = this.checked;
+    $(':checkbox').each(function() {
+      this.checked = selectAllChecked;
+    });
+  });
+
+  $("#autogroups-back-button").on('click', function () {
+    $('#wizardAction').val('back');
+    $('#autogroups-wizard-step2-form').submit();
+  });
+  /*]]>*/
+</script>
+
+<script th:fragment="autoGroupsStep3Js" th:inline="javascript">
+  /*<![CDATA[*/
+  $("input[name='structureConfigurationOption']").change(function(){
+    this.value === '1' ? $('#mixtureStructureOptions').fadeOut() : $('#mixtureStructureOptions').fadeIn();
+  });
+
+  $("input[name='splitOptions']").change(function(){
+    var groupSelectedClass = 'optGroupSelectSelected';
+    if (this.value === '1') {
+      $('#splitByGroups').removeClass(groupSelectedClass);
+      $('#splitByUsers').addClass(groupSelectedClass);
+      $('#splitByGroupsOptions').fadeOut();
+      $('#splitByUsersOptions').fadeIn();
+    } else {
+      $('#splitByUsers').removeClass(groupSelectedClass);
+      $('#splitByGroups').addClass(groupSelectedClass);
+      $('#splitByUsersOptions').fadeOut();
+      $('#splitByGroupsOptions').fadeIn();
+    }
+  });
+
+  $("#autogroups-back-button").on('click', function () {
+    $('#wizardAction').val('back');
+    $('#autogroups-wizard-step3-form').submit();
+  });
+  /*]]>*/
+</script>
+
+<script th:fragment="autoGroupsStep4Js" th:inline="javascript">
+  /*<![CDATA[*/
+  $("#autogroups-back-button-top,#autogroups-back-button").on('click', function () {
+    $('#wizardAction').val('back');
+    $('#autogroups-wizard-step4-form').submit();
+  });
+  /*]]>*/
+</script>
```

Comment:
Generally, do you need to keep all this code in a thymeleaf template? Why not just have it in a js file and include that like normal. The benefit of doing that is that you can modify js code directly in your deployed webapp. Rapid iteration on your js, all that good stuff. This way, you have to compile your app each time.

Reply:
I don't like to have a JS file loaded in every template, to be honest, I like to load small portions of JS with the required and needed JS for each template. for example, if I load tons of webjars for several templates, all the webjars are loaded even if they are not needed, I prefer to use fragments instead but I see your point.

Reply:
I don't like to have a JS file loaded in every template, to be honest, I like to load small portions of JS with the required and needed JS for each template. for example, if I load tons of webjars for several templates, all the webjars are loaded even if they are not needed, I prefer to use fragments instead but I see your point.